samples: Bluetooth: df: Forward GPIO pins from app to network core

ppryga-nordic opened the pull request:
To give control over GPIO pins for Direction Finding Extension of
Radio peripheral when build for nRF53 network core, the application
core has to assign those pins to networ code.

There is a mechanism that uses a device tree overlay to get
information about GPIO pins to be assigned to network core.

The commit adds overlays with appropriate configuration
to assign GPIO pins in all DF related samples.

This PR depends on: https://github.com/zephyrproject-rtos/zephyr/pull/42576/files

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga-nordic committed:
samples: Bluetooth: df: Forward GPIO pins from app to network core

To give control over GPIO pins for Direction Finding Extension of
Radio peripheral when build for nRF53 network core, the application
core has to assign those pins to network core.

There is a mechanism that uses a device tree overlay to get
information about GPIO pins to be assigned to network core.

The commit adds overlays with appropriate configuration
to assign GPIO pins in all DF related samples.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

Thalley commented on samples/bluetooth/direction_finding_central/boards/nrf5340dk_nrf5340_cpuapp.overlay:
```diff
@@ -0,0 +1,20 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ /* Enable pin forwarding to network code. The selected pins will be used by
```
I'm not familiar with the term `network code` - Can you elaborate?

Thalley replied:
Reading further below: Did you mean `network core`?

Thalley replied:
Applies multiple places

Thalley replied:
And in the commit message :)

ppryga replied:
ah, thanks for spotting that. It should be `network core` everywhere. I assume that there is no need to elaborate about the topic. In case there is such need write it here please.

Comments and commit corrected.

cvinayak commented on samples/bluetooth/direction_finding_central/README.rst:
```diff
@@ -45,6 +45,9 @@ enabled:
   :zephyr_file:`samples/bluetooth/direction_finding_central/boards/nrf52833dk_nrf52833.overlay`
   to a new file,
   :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
+* Make sure the same GPIO pins are assigned to Direction Finding Extension in file
+  :zephyr_file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpuapp.overlay`.
+  as the one in created file :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
```
```suggestion
* Make sure the same GPIO pins are assigned to Direction Finding Extension in file
  :zephyr_file:`samples/bluetooth/direction_finding_central/boards/nrf5340dk_nrf5340_cpuapp.overlay`.
  as the one in created file :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
```

ppryga replied:
Corrected

cvinayak commented on samples/bluetooth/direction_finding_connectionless_rx/README.rst:
```diff
@@ -45,6 +45,9 @@ enabled:
   :zephyr_file:`samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52833.overlay`
   to a new file,
   :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
+* Make sure the same GPIO pins are assigned to Direction Finding Extension in file
+  :zephyr_file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpuapp.overlay`.
+  as the one in created file :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
```
```suggestion
* Make sure the same GPIO pins are assigned to Direction Finding Extension in file
  :zephyr_file:`samples/bluetooth/direction_finding_connectionless_rx/boards/nrf5340dk_nrf5340_cpuapp.overlay`.
  as the one in created file :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
```

ppryga replied:
Corrected

cvinayak commented on samples/bluetooth/direction_finding_connectionless_tx/README.rst:
```diff
@@ -45,6 +45,9 @@ support enabled:
   :zephyr_file:`samples/bluetooth/direction_finding_connectionless_tx/boards/nrf52833dk_nrf52833.overlay`
   to a new file,
   :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
+* Make sure the same GPIO pins are assigned to Direction Finding Extension in file
+  :zephyr_file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpuapp.overlay`.
+  as the one in created file :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
```
```suggestion
* Make sure the same GPIO pins are assigned to Direction Finding Extension in file
  :zephyr_file:`samples/bluetooth/direction_finding_connectionless_tx/boards/nrf5340dk_nrf5340_cpuapp.overlay`.
  as the one in created file :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
```

ppryga replied:
Corrected

cvinayak commented on samples/bluetooth/direction_finding_peripheral/README.rst:
```diff
@@ -44,6 +44,9 @@ enabled:
   :zephyr_file:`samples/bluetooth/direction_finding_peripheral/boards/nrf52833dk_nrf52833.overlay`
   to a new file,
   :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
+* Make sure the same GPIO pins are assigned to Direction Finding Extension in file
+  :zephyr_file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpuapp.overlay`.
+  as the one in created file :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
```
```suggestion
* Make sure the same GPIO pins are assigned to Direction Finding Extension in file
  :zephyr_file:`samples/bluetooth/direction_finding_peripheral/boards/nrf5340dk_nrf5340_cpuapp.overlay`.
  as the one in created file :file:`samples/bluetooth/hci_rpmsg/boards/nrf5340dk_nrf5340_cpunet.overlay`.
```

ppryga replied:
Corrected

carlescufi commented:
@ppryga please rebase to the latest `main`

ppryga-nordic committed:
samples: Bluetooth: df: add missing integration platforms

Add missing integration plaforms in direction finding samples.
There were no nRF52820 and nRF5340 added.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>
